msteams-platform/tabs/how-to/dashboard-teams-toolkit.md

```diff
@@ -0,0 +1,991 @@
+
+1. Add the widget to dashboard layout. Include the new widget in your dashboard file:
+
+```typescript
+...
+export default class YourDashboard extends Dashboard {
+  ...
+  protected dashboardLayout(): undefined | JSX.Element {
+    return (
+      <>
+        <GraphWiget />
+      </>
+    );
+  }
+  ...
+}
+```
+
+Now, launch or refresh your Teams app, you'll see the new widget using Microsoft Graph Toolkit.
+
+### How to include a data loader
+
+If you want to include a data loader to your widget before the widget is loaded, you can add a property to the state of the widget to indicate that the data loader is loading. This property can be used to show a loading indicator to the user.
+
+The following steps show how to add a property to the state of `ListWidget` and how to use it to show a loading spinner while the data is loading.
+
+1. **Define a state type**
+
+   Define a state type including a property named `loading` that indicates whether the data is loading.
+
+    ```typescript
+    interface ListWidgetState {
+      data: ListModel[];
+      loading?: boolean;
+    }
+    ```
+
+1. **Add a data loader**
+
+    Modify the `bodyContent` method to show a loading spinner if data is loading.
+
+    ```typescript
+    bodyContent(): JSX.Element | undefined {
+      return (
+        <>
+          {this.state.loading !== false ? (
+            <div style={{ display: "grid", justifyContent: "center", height: "100%" }}>
+              <Spinner label="Loading..." labelPosition="below" />
+            </div>
+          ) : (
+            <div style={bodyContentStyle()}>
+              ...
+            </div>
+          )}
+        </>
+      );
+    }
+    ```
+
+1. **Hide the footer button if the data is loading**
+
+    Following is an example of footer button:
+
+    ```typescript
+    footerContent(): JSX.Element | undefined {
+      if (this.state.loading === false) {
+        return (
+          <Button
+            ...
+          </Button>
+        );
+      }
+    }
+    ```
+
+1. **Update the state reference**
+
+    Update the state reference in the widget file to use the new state type and update the state in the `getData` method to set the `loading` property to `false` after the data is loaded.
+
+    Now, the loading spinner is shown while the data is loading. When the data is loaded, the loading spinner is hidden, and the list data and footer button are shown.
+
+    :::image type="content" source="../../assets/images/sbs-create-a-new-dashboard/spinner.gif" alt-text="Graphical representation shows the loading spinner while the data is loading.":::
+
+### How to handle empty state
+
+You can display a specific content in your widget when the data is empty. To do so, you need to modify the `bodyContent` method in your widget file to adopt different states of the data. The following example shows how to display an empty image when the data of `ListWidget` is empty.
+
+```typescript
```

Comment:
It may be better if the language of this code snippet is adjusted to `tsx`.

Reply:
Updated to `tsx`